src/auth/cephx/CephxServiceHandler.cc

```diff
@@ -183,8 +183,9 @@ int CephxServiceHandler::handle_request(bufferlist::iterator& indata, bufferlist
     {
       ldout(cct, 10) << "handle_request getting rotating secret for " << entity_name << dendl;
       build_cephx_response_header(cephx_header.request_type, 0, result_bl);
-      key_server->get_rotating_encrypted(entity_name, result_bl);
-      ret = 0;
+      ret = key_server->get_rotating_encrypted(entity_name, result_bl);
+      if (!ret)
+        return -EPERM;
```

Comment:
We've had several issues from conflating integer and boolean return signatures. Let's not exacerbate those by assigning from a boolean into an int!